[WIP] stat page

slashmili opened the pull request.

slashmili committed:
Added stat base schema

slashmili committed:
Added elink_id as part of required fields

carloslima commented on priv/repo/migrations/20160226131844_create_stat.exs:
```diff
@@ -0,0 +1,22 @@
+      add :tag, :stats_tags
+      add :value, :string
+      add :count, :integer
+      add :visited_at, :datetime
```
how do you want to handle visited_at on an aggregated row?

slashmili replied:
Not sure yet! Looking for an option, PG 9.5 has [UPSERT](http://www.postgresql.org/docs/devel/static/sql-insert.html) Ecto doesn't support it yet.

I'm thinking that:
- we can have a pool of stats which update the counter one a at time for each tag
- Execute  `ON CONFLICT DO UPDATE`

slashmili committed:
Added unique index for stats key column

slashmili committed:
Added missing column(:visiting_interval) to stats unique index

slashmili committed:
Refactored test

slashmili committed:
Implemented a rough idea of how StatServer should work

slashmili committed:
Added Timex and Timex.Ecto to handle dates easier

slashmili committed:
Used Timex in StatServer

Timex has helper functions that helps to deal with date and time

slashmili committed:
Create multiple stats record based on passed tag

slashmili committed:
Used current datetime as default value for visited_at

slashmili committed:
Moved StatServer to StatWorker in preparation for a Pool of workers

slashmili committed:
Using pool of StatWorker to save elink visit data

slashmili committed:
Implemented a simple browser detection

See more https://developer.mozilla.org/en-US/docs/Browser_detection_using_the_user_agent

slashmili committed:
Exclude test that require postgres 9.5 in travis

slashmili committed:
Added country detection library to elide

slashmili committed:
Fixed the test

slashmili committed:
Added support for extracting domain to Stat

slashmili committed:
Increase stat on each elink visit

slashmili committed:
Set the visit time on client side

There is no grantee that the message in mail box is going to be proceed
immediately. So I moved the inserting date to the client side.

slashmili committed:
Added url link to stat so we can get details of vist per Elink and Urls

slashmili committed:
Merge remote-tracking branch 'origin/develop' into stat-page
